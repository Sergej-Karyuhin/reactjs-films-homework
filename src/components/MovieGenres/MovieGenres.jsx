import React from 'react';
import styles from './MovieGenres.scss';

const MovieGenres = ({ movieGenres }) => (
  <ul className={styles.genres}>
    {
      movieGenres.map( (genre, index) =>
        <li className={styles.genre} key={ index }>
          { genre }
        </li>
      )
    }
  </ul>
);

export default MovieGenres;
