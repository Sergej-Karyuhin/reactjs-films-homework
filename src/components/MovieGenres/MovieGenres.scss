.genres {
  display: flex;

  .genre {
    margin-right: 10px;
    font-size: 18px;
  }
}
