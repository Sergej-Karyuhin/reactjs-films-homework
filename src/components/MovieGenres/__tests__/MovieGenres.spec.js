import React from 'react';
import renderer from 'react-test-renderer';
import MovieGenres from '../MovieGenres';

const genres = ['Adventure', 'Drama', 'Family', 'Fantasy'];

it('renders correctly', () => {
  const tree = renderer
    .create(<MovieGenres movieGenres={genres} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
