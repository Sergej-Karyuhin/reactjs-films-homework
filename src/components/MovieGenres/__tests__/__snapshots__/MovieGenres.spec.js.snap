// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="genres"
>
  <li
    className="genre"
  >
    Adventure
  </li>
  <li
    className="genre"
  >
    Drama
  </li>
  <li
    className="genre"
  >
    Family
  </li>
  <li
    className="genre"
  >
    Fantasy
  </li>
</ul>
`;
